docs/resources/cse_microservice.md

```diff
@@ -0,0 +1,112 @@
+---
+subcategory: "Cloud Service Engine (CSE)"
+---
+
+# huaweicloud_cse_microservice
+
+Manages a dedicated microservice resource within HuaweiCloud.
+
+-> When deleting a microservice, all instances under it will also be deleted together.
+
+## Example Usage
+
+### Create a microservice in an engine with RBAC authentication disabled
+
+```hcl
+variable "engine_conn_addr" {}
+variable "service_name" {}
+variable "app_name" {}
+
+resource "huaweicloud_cse_microservice" "test" {
+  connect_address = var.engine_conn_addr
+  name            = var.service_name
+  verison         = "1.0.0"
+  environment     = "development"
+  app_name        = var.app_name
+}
+```
+
+### Create a microservice in an engine with RBAC authentication enabled
+
+```hcl
+variable "engine_conn_addr" {}
+variable "service_name" {}
+variable "app_name" {}
+
+resource "huaweicloud_cse_microservice" "test" {
+  connect_address = var.engine_conn_addr
+  name            = var.service_name
+  verison         = "1.0.0"
+  environment     = "development"
+  app_name        = var.app_name
+
+  admin_user = "root"
+  admin_pass = "Huawei!123"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `connect_address` - (Required, String, ForceNew) Specifies the connection address of service registry center for the
+  specified dedicated CSE engine. Changing this will create a new microservice.
+
+* `name` - (Required, String, ForceNew) Specifies the name of the dedicated microservice.
+  The name can contain `1` to `128` characters, only letters, digits, underscore (_), hyphens (-) and dots (.) are
+  allowed. The name must start and end with a letter or digit. Changing this will create a new microservice.
+
+* `app_name` - (Required, String, ForceNew) Specifies the name of the dedicated microservice application.
+  Changing this will create a new microservice.
+
+* `version` - (Required, String, ForceNew) Specifies the version of the dedicated microservice.
+  Changing this will create a new microservice.
+
+* `environment` - (Optional, String, ForceNew) Specifies the environment (stage) type.
+  The valid values are **development**, **testing**, **acceptance** and **production**.
+  If omitted, the microservice will be deployed in an empty environment.
+  Changing this will create a new microservice.
+
+* `level` - (Optional, String, ForceNew) Specifies the microservice level.
+  The valid values are **FRONT**, **MIDDLE**, and **BACK**. Changing this will create a new microservice.
+
+* `description` - (Optional, String, ForceNew) Specifies the description of the dedicated microservice.
+  The description can contian a maximum of `256` characters.
+  Changing this will create a new microservice.
+
+* `status` - (Optional, String, ForceNew) Specifies the microservice status. The valid values are **UP** and **DOWN**.
```

Comment:
forceNew will delete the instance, if cannot update the status,  change to computed attr is better.

Reply:
done.